src/xtt.h

```diff
@@ -49,6 +54,7 @@ enftun_xtt_handshake(const char** server_hosts,
                      const char* socket_port,
                      const char* ca_cert_file,
                      const char* basename,
-                     struct enftun_xtt* xtt);
+                     struct enftun_xtt* xtt,
+                     struct enftun_tls* tls);
```

Comment:
Don't mix the `enftun_tls` socket from the `enftun_context` into `enftun_xtt`.  XTT doesn't use TLS, so it shouldn't have a dependency on our TLS code.

There's no need to remember which mode (native vs SCM) was used by XTT and use that for the TLS tunnel as well.  XTT is normally only run once in a devices lifetime, so we'd only be saving a few seconds once.

Reply:
Removed.